src/beignet/_bisect.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import Callable, Literal
+
+import torch
+from torch import Tensor
+
+from beignet._root_scalar import RootSolutionInfo
+
+
+def bisect(
+    f: Callable,
+    *args,
+    lower: float,
```

Comment:
Use `a` and the type should be loosened to `float | Tensor`.

Reply:
upper/lower seem a bit more descriptive. Longer argument names for keyword arguments seem ok to me. Do we have an established convention?